File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> m;
        int sum = 0;
        int count = 0;

        m[0] = 1;

        for(int i = 0; i < nums.size(); i++){
            sum += nums[i];
            int r=sum-k;
            count+=m[r];
            m[sum]+=1;
        }

        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer,Integer>mpp=new HashMap<>();
        int presum=0;
        int count=0;
        mpp.put(0,1);
        for(int i=0;i<nums.length;i++){
            presum+=nums[i];
            int ke=presum-k;
            count+=mpp.getOrDefault(ke,0);
            mpp.put(presum,mpp.getOrDefault(presum,0)+1);
            
        }
        return count;
    }
}